File: src/web/my-app/src/API/testeos.js
// API/Testeos.js
export class testeos {

//   constructor() {}

  // Valida que el nombre de la emisora exista y sea igual al esperado
  testNombreEmisora(nombre, esperado = "LE_WH-1000XM5") {
    
    if (!nombre) {
      console.error("❌ Error: nombre de emisora es null o undefined");
      return false;
    }

    if (nombre === esperado) {
      console.log(`✅ Test correcto: ${nombre} coincide con ${esperado}`);
      return true;
    } else {
      console.error(`❌ Test fallido: ${nombre} no coincide con ${esperado}`);
      return false;
    }
  }
}

File: src/AndroidBiometria/app/src/main/java/com/example/androidbiometria/Testeos.java
package com.example.androidbiometria;

import android.util.Log;

public class Testeos {

    private static final String TAG = "test";

    /**
     * Test simple para verificar que Firebase está inicializado.
     */
    public static void testFirebase(EnviarDatosDeIBeacon enviarDatos) {
        if (enviarDatos != null) {
            Log.d(TAG, "✅ Firebase inicializado correctamente y objeto EnviarDatosDeIBeacon listo");
        } else {
            Log.e(TAG, "❌ Error: EnviarDatosDeIBeacon es null");
        }
    }

    /**
     * Test opcional para validar dispositivo detectado.
     */
    public static boolean testFiltrarDispositivo(String nombreDetectado) {
        String dispositivoBuscado = "LE_WH-1000XM5";

        Log.d(TAG, "🔍 Probando filtro de dispositivo...");
        Log.d(TAG, "Esperando: " + dispositivoBuscado);
        Log.d(TAG, "Detectado: " + nombreDetectado);

        boolean coincidencia = dispositivoBuscado.equals(nombreDetectado);

        if (coincidencia) {
            Log.d(TAG, "✅ Test correcto: se detectó el dispositivo esperado (" + dispositivoBuscado + ")");
        } else {
            Log.e(TAG, "❌ Test fallido: no coincide con el esperado");
        }

        return coincidencia;
    }

    /**
     * Método para enviar a Firebase desde el testeo.
     */
    public static void testEnviarAFirebase(String nombreDetectado, EnviarDatosDeIBeacon enviarDatos) {
        if (enviarDatos != null && nombreDetectado != null) {
            enviarDatos.enviarNombreDeEmisora(nombreDetectado);
            Log.d(TAG, "✅ Test correcto: Nombre de emisora enviado a Firebase");
        } else {
            Log.e(TAG, "❌ No se puede enviar a Firebase, objeto o nombre null");
        }
    }
}
